home.nix

```diff
@@ -29,7 +29,12 @@ in
 {
   options.programs.direnv-instant = {
     enable = mkEnableOption "non-blocking direnv integration daemon with tmux support";
-    package = mkPackageOption pkgs "direnv-instant" { };
+    package = mkOption {
+      type = package;
+      default = pkgs.callPackage ./default.nix { };
+      defaultText = lib.literalExpression "pkgs.callPackage ./default.nix { }";
+      description = "The direnv-instant package to use.";
+    };
```

Comment:
I think this is very unconventional but ok. An overlay could still be useful as a flake output though.

Reply:
Package option are a quiet common pattern in NixOS. Overlays add a significant of evaluation overhead due to the fix point lookup and are also not introspectable/hard to debug, so I avoid them. When having a lot of machines it adds up, because nixpkgs.overlays forces one to have one nixpkgs per machine rather than beeing able to one shared nixpkgs instance via nixpkgs.pkgs (https://zimbatm.com/notes/1000-instances-of-nixpkgs)